tests/integration/client-id-generation-test.js

```diff
@@ -1,101 +1,116 @@
 import { resolve } from 'rsvp';
 import { get } from '@ember/object';
-import { run } from '@ember/runloop';
-import setupStore from 'dummy/tests/helpers/store';
-
+import { setupTest } from 'ember-qunit';
 import { module, test } from 'qunit';
+import Model from 'ember-data/model';
+import JSONAPIAdapter from 'ember-data/adapters/json-api';
+import { attr, belongsTo, hasMany } from '@ember-decorators/data';
+
+module('integration - Client Id Generation', function(hooks) {
+  setupTest(hooks);
+  let store;
+  let adapter;
+
+  hooks.beforeEach(function() {
+    let { owner } = this;
+
+    class Comment extends Model {
+      @attr
+      text;
+      @belongsTo('post', { async: false, inverse: 'comments' })
+      post;
+    }
+    class Post extends Model {
+      @attr
+      title;
+      @hasMany('comment', { async: false, inverse: 'post' })
+      comments;
+    }
+    class Misc extends Model {
+      @attr('string')
+      foo;
+    }
 
-import DS from 'ember-data';
-
-let Post, Comment, Misc, env;
-
-module('integration/client_id_generation - Client-side ID Generation', {
-  beforeEach() {
-    Comment = DS.Model.extend({
-      post: DS.belongsTo('post', { async: false }),
-    });
-
-    Post = DS.Model.extend({
-      comments: DS.hasMany('comment', { async: false }),
-    });
-
-    Misc = DS.Model.extend({
-      foo: DS.attr('string'),
-    });
-
-    env = setupStore({
-      post: Post,
-      comment: Comment,
-      misc: Misc,
-    });
-  },
-
-  afterEach() {
-    run(env.container, 'destroy');
-  },
-});
+    owner.register('model:comment', Comment);
+    owner.register('model:post', Post);
+    owner.register('model:misc', Misc);
+    owner.register('adapter:application', JSONAPIAdapter);
 
-test('If an adapter implements the `generateIdForRecord` method, the store should be able to assign IDs without saving to the persistence layer.', function(assert) {
-  assert.expect(6);
+    store = owner.lookup('service:store');
+    adapter = store.adapterFor('application');
+  });
 
-  let idCount = 1;
+  test('If an adapter implements the `generateIdForRecord` method, the store should be able to assign IDs without saving to the persistence layer.', async function(assert) {
+    assert.expect(6);
+
+    let idCount = 1;
+
+    adapter.generateIdForRecord = function(passedStore, record) {
+      assert.ok(store === passedStore, 'store is the first parameter');
+
+      return 'id-' + idCount++;
+    };
+
+    adapter.createRecord = function(store, modelClass, snapshot) {
+      let type = modelClass.modelName;
+
+      if (type === 'comment') {
+        assert.equal(snapshot.id, 'id-1', "Comment passed to `createRecord` has 'id-1' assigned");
+        return resolve({
+          data: {
+            type,
+            id: snapshot.id,
+          },
+        });
+      } else {
+        assert.equal(snapshot.id, 'id-2', "Post passed to `createRecord` has 'id-2' assigned");
+        return resolve({
```

Comment:
I feel such tests would belong being in a set of tests explicitly testing how we behave around "in-flight-attributes". I default to tests that return payloads for the general case as this will prevent unnecessary refactoring later should we decide to change how we handle in-flight.